doc/languages-frameworks/neovim.section.md

```diff
@@ -1,22 +1,40 @@
 # Neovim {#neovim}
 
 Install `neovim-unwrapped` to get a barebone neovim to configure imperatively.
-Neovim can be configured to include your favorite plugins and additional libraries by installing `neovim` instead.
+This is the closest to what you encounter on other distributions.
+
+`neovim` is a wrapper around neovim with some extra configuration to for
+instance set the various language providers like python.
+The wrapper can be further configured to include your favorite plugins and
+configurations for a reproducible neovim across machines.
 See the next section for more details.
 
 ## Custom configuration {#neovim-custom-configuration}
 
-For Neovim the `configure` argument can be overridden to achieve the same:
+There are two wrappers available to provide additionnal configuration around the vanilla package `pkgs.neovim-unwrapped`:
+1. `wrapNeovim`: the historical one you should use
+2. `wrapNeovimUnstable` intended to replace the former. It has more features but
+   the interface is not stable yet.
+
+You can configure the former via:
 
 ```nix
 neovim.override {
   configure = {
     customRC = ''
       # here your custom configuration goes!
     '';
+    packages.myVimPackage = with pkgs.vimPlugins; {
+      # see examples below how to use custom packages
+      start = [ ];
+      # If a Vim plugin has a dependency that is not explicitly listed in
+      # opt that dependency will always be added to start to avoid confusion.
```

Comment:
```suggestion
      # If a Vim plugin has a dependency that is not explicitly listed in
      # `opt`, that dependency will always be added to `start` to avoid confusion.
```